awscli/argprocess.py

```diff
@@ -379,6 +379,16 @@ def _handle_special_cases(self, cli_argument, value, service_id,
             # {"Value": "shutdown"}
             return {'Value': value}
 
+        # Additional handling of special cases where a structure is
+        # a document type. Shorthand syntax is not supported and
+        # should therefore err out before attempt to parse.
+        if model.type_name == 'structure' and model.is_document_type:
```

Comment:
Why no shorthand support for document types? What if the value the user needs to provide are scalars like `"string"` or `true` or `10`?